lib/player.js

```diff
@@ -35,6 +35,7 @@ goog.require('shaka.media.StreamingEngine');
 goog.require('shaka.media.TimeRangesUtils');
 goog.require('shaka.net.NetworkingEngine');
 goog.require('shaka.net.NetworkingUtils');
+goog.require('shaka.Player.ErrorEvent');
```

Comment:
I think the errors are caused by ordering and namespaces here.  goog.require here will assign the ErrorEvent class to shaka.Player.ErrorEvent.  Later in this file, we assign shaka.Player = class ..., overwriting what was assigned to ErrorEvent by replacing the whole shaka.Player namespace.

So we might need to do something like this:

1. Rename shaka.Player.ErrorEvent to shaka.errors.ErrorEvent, giving errors their own namespace
2. ~For backward compatibility, assign shaka.Player.ErrorEvent = shaka.errors.ErrorEvent after the definition of shaka.Player~
3. Actually, no need for backward compatibility.  That class never existed at runtime, and was also never in our externs.  It only ever existed as a doc, so it's fine to rename it AFAICT.  But the new name should be exported so it's in the generated externs/defs, where other projects can refer to it as a strong type.

Reply:
Done, but I used `shaka.events.ErrorEvent` instead